lib/config-generator.js

```diff
@@ -0,0 +1,261 @@
+     */
+    calc() {
+        const isESMModule = isPackageTypeModule(this.packageJsonPath);
+
+        this.result.configFilename = isESMModule ? "eslint.config.js" : "eslint.config.mjs";
+
+        let importContent = "";
+        const helperContent = `import path from "path";
+import { fileURLToPath } from "url";
+import { FlatCompat } from "@eslint/eslintrc";
+import pluginJs from "@eslint/js";
+
+// mimic CommonJS variables -- not needed if using CommonJS
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({baseDirectory: __dirname, recommendedConfig: pluginJs.configs.recommended});
+`;
+        let exportContent = "";
+        let needCompatHelper = false;
+
+        if (this.answers.moduleType === "commonjs" || this.answers.moduleType === "script") {
+            exportContent += `  {files: ["**/*.js"], languageOptions: {sourceType: "${this.answers.moduleType}"}},\n`;
+        }
+
+        if (this.answers.purpose === "syntax") {
+
+            // no need to install any plugin
+        } else if (this.answers.purpose === "problems") {
+            this.result.devDependencies.push("@eslint/js");
+            importContent += "import pluginJs from \"@eslint/js\";\n";
+            exportContent += "  pluginJs.configs.recommended,\n";
+        } else if (this.answers.purpose === "style") {
+            const styleguide = typeof this.answers.styleguide === "string"
+                ? { packageName: this.answers.styleguide, type: "flat" }
+                : this.answers.styleguide;
+
+            this.result.devDependencies.push(styleguide.packageName);
+
+            // install peer dependencies - it's needed for most eslintrc-style shared configs.
+            const peers = fetchPeerDependencies(styleguide.packageName);
+
+            if (peers !== null) {
+                this.result.devDependencies.push(...peers);
+            }
+
+            if (styleguide.type === "flat" || styleguide.type === void 0) {
+                importContent += `import styleGuide from "${styleguide.packageName}";\n`;
+                exportContent += "  ...[].concat(styleGuide),\n";
+            } else if (styleguide.type === "eslintrc") {
+                needCompatHelper = true;
+
+                const shorthandName = getShorthandName(styleguide.packageName, "eslint-config");
+
+                exportContent += `  ...compat.extends("${shorthandName}"),\n`;
+            }
+        }
+
+        if (this.answers.env?.length > 0) {
+            this.result.devDependencies.push("globals");
+            importContent += "import globals from \"globals\";\n";
+            const envContent = {
+                browser: "globals: globals.browser",
+                node: "globals: globals.node",
+                "browser,node": "globals: {...globals.browser, ...globals.node}"
+            };
+
+            exportContent += `  {languageOptions: ${envContent[this.answers.env.join(",")]}},\n`;
```

Comment:
```suggestion
            exportContent += `  {languageOptions: { ${envContent[this.answers.env.join(",")]} }},\n`;
```

The snapshots must be also fixed.